protocol/identity/alias/generate.go

```diff
@@ -32,16 +30,31 @@ func GenerateFromPublicKey(publicKey *ecdsa.PublicKey) string {
 // GenerateFromPublicKeyString returns the 3 words name given a public key
 // prefixed with 0x
 func GenerateFromPublicKeyString(publicKeyString string) (string, error) {
-	pk := strings.TrimPrefix(publicKeyString, "0x")
-	publicKeyBytes, err := hex.DecodeString(pk)
-	if err != nil {
-		return "", err
+	var fixedPublicKey string
+	if len(publicKeyString) == multiformat.LegacyKeyLength {
+		fixedPublicKey = publicKeyString
+	} else {
+		// Attempt to fix a maybe public key without the 0x prefix
+		if len(publicKeyString) >= 2 && publicKeyString[:2] != "0x" {
+			fixedPublicKey = "0x" + publicKeyString
+		} else {
+			return "", fmt.Errorf("invalid length of public key")
+		}
```

Comment:
I think the order of checks should be like this:
```suggestion
    // Attempt to fix a maybe public key without the 0x prefix
    if len(publicKeyString) >= 2 && publicKeyString[:2] != "0x" {
	    fixedPublicKey = "0x" + publicKeyString
    }
	if len(publicKeyString) != multiformat.LegacyKeyLength {
	    return fmt.Errorf("invalid length of public key")
	}
```

Because in the end the length of the string must be 132 and start with `0x`.
Otherwise one could pass a 1000 characters string, and get no error here.

Reply:
A test for this would be nice as well 🙏